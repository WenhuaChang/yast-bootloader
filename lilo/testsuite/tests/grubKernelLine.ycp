/**
 * File:
 *  grubKernelLine.ycp
 *
 * Module:
 *  Bootloader configurator
 *
 * Summary:
 *  Kernel parameters parsing/patching tests.
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    import "Pkg";

    // minimal "don't care" SCR data to work with the constructor
    map READ = $[
	"target": $[
	    "tmpdir": "/tmp",
	    ],
	"etc" : $[
            "install_inf" : $[
                "Locale" : "en_EN",
                "InstMode" : "cd",
		],
	    ],
	"sysconfig" : $[
            "console" : $[
                "CONSOLE_FONT" : "lat2-16.psfu",
                "CONSOLE_SCREENMAP" : "trivial",
                "CONSOLE_UNICODEMAP" : "",
                "CONSOLE_MAGIC" : "(K",
                "CONSOLE_ENCODING" : "ISO-8859-2",
            ],
        ],
        "probe" : $[
            "disk" : [],
            "boot_arch" : "lilo",
            "architecture" : "i386",
            "has_apm" : true,
            "has_pcmcia" : false,
            "has_smp" : false,
            "system" : [],
            "floppy" : $[
                "manual" : [],
		],
	    "cdrom" : $[
                "manual" : [],
		],
	    ],
	];
    TESTSUITE_INIT ([READ, $[], $[]], 0);

    include "lilo/routines.ycp";

        READ = $["grub": $["sections" : $["linux" : $["kernel" : "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333"]]]];

    TEST (``(getGrubKernelParamFromLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "root")), [READ, $[], $[]], 0);
    TEST (``(getGrubKernelParamFromLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "vga")), [READ, $[], $[]], 0);
    TEST (``(getGrubKernelParamFromLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "unex")), [READ, $[], $[]], 0);
    TEST (``(getGrubKernelParamFromLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "noval")), [READ, $[], $[]], 0);
    DUMP("==================================================================");
    TEST (``(setGrubKernelParamToLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "root", "/dev/hda5")), [READ, $[], $[]], 0);
    TEST (``(setGrubKernelParamToLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "vga", "55")), [READ, $[], $[]], 0);
    TEST (``(setGrubKernelParamToLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "ide", "nodma")), [READ, $[], $[]], 0);
    TEST (``(setGrubKernelParamToLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "vga", "false")), [READ, $[], $[]], 0);
    TEST (``(setGrubKernelParamToLine ("(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333", "extra", "true")), [READ, $[], $[]], 0);

}

