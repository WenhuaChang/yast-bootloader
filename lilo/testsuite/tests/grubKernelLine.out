Return	/dev/hda3
Return	333
Return	false
Return	true
Dump	==================================================================
Return	(hd0,0)/vmlinuz.suse noval root=/dev/hda5 vga=333
Return	(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=55
Return	(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma
Return	(hd0,0)/vmlinuz.suse noval root=/dev/hda3
Return	(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 extra
